palette/src/rgb/mod.rs

```diff
@@ -70,3 +71,75 @@ pub trait Primaries: Any {
     ///Primary blue.
     fn blue<Wp: WhitePoint, T: Component + Float>() -> Yxy<Wp, T>;
 }
+
+impl<T, U> From<LinSrgb<T>> for Srgb<U>
+where
+    T: Component + Float,
+    U: Component,
+{
+    fn from(lin_srgb: LinSrgb<T>) -> Self {
+        let non_lin = Srgb::<T>::from_linear(lin_srgb);
+        non_lin.into_format()
+    }
+}
+
+impl<T, U> From<Srgb<T>> for LinSrgb<U>
+where
+    T: Component + Float,
+    U: Component,
+{
+    fn from(srgb: Srgb<T>) -> Self {
+        srgb.into_linear()
+            .into_format()
+    }
+}
+
+impl<T, U> From<LinSrgb<T>> for Srgba<U>
+where
+    T: Component + Float,
+    U: Component,
+{
+    fn from(lin_srgb: LinSrgb<T>) -> Self {
+        let color = lin_srgb.into();
+        let alpha = U::max_intensity();
+        Alpha { color, alpha }
```

Comment:
Hmm... You should not have to construct the `Alpha` like this. You should be able to use `from_linear` and `into_format` on these too. Otherwise that's also a hole in the API.